src/utils/helpers.ts

```diff
@@ -0,0 +1,104 @@
+import { Person, SearchParams, SortType } from '../types';
+
+export function getSearchWith(
+  currentParams: URLSearchParams,
+  paramsToUpdate: SearchParams,
+): string {
+  const newParams = new URLSearchParams(
+    currentParams.toString(),
+  );
+
+  Object.entries(paramsToUpdate)
+    .forEach(([key, value]) => {
+      if (value === null) {
+        newParams.delete(key);
+      } else if (Array.isArray(value)) {
+        newParams.delete(key);
+
+        value.forEach(part => {
+          newParams.append(key, part);
+        });
+      } else {
+        newParams.set(key, value);
+      }
+    });
+
+  return newParams.toString();
+}
+
+export const getParent = (people: Person[], name: string | null) => (
+  people.find(person => person.name === name)
+);
+
+export const filterPeople = (
+  people: Person[],
+  sex: string,
+  query: string,
+  centuries: string[],
+) => {
+  let filteredPeople = [...people];
+
+  if (sex) {
+    filteredPeople = filteredPeople
+      .filter(({ sex: personSex }) => personSex === sex);
+  }
+
+  if (query) {
+    filteredPeople = filteredPeople.filter(({
+      name,
+      motherName,
+      fatherName,
+    }) => {
+      const availableResults = (name + motherName + fatherName).toLowerCase();
+
+      return availableResults.includes(query.toLowerCase());
```

Comment:
bad idea, its better to check them separate, because for example if i write last chars of person name and then first chars of persons mother name, it will show me that person, but its inccorect
![image](https://user-images.githubusercontent.com/122156399/233075823-c549bfa0-83ea-410f-b69c-87193046b645.png)